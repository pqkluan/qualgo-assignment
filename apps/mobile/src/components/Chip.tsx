import { FC } from 'react';
import { View } from 'react-native';
import { createStyleSheet, useStyles } from 'react-native-unistyles';

import { Text } from '@mobile/components/Text';

type Props = {
	text: string;
};

export const Chip: FC<Props> = (props) => {
	const { text } = props;

	const { styles } = useStyles(stylesheet);

	return (
		<View style={styles.container}>
			<Text style={styles.text}>{text}</Text>
		</View>
	);
};

const stylesheet = createStyleSheet((theme) => ({
	container: {
		backgroundColor: theme.colors.typography,
		paddingHorizontal: theme.margins.sm,
		paddingVertical: theme.margins.xs,
		borderRadius: theme.radius.xs,
	},
	text: {
		color: theme.colors.background,
	},
}));
