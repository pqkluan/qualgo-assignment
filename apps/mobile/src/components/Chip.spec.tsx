import React from 'react';
import { render } from '@testing-library/react-native';

import { Chip } from './Chip';

describe('Chip', () => {
	it('renders chip text', () => {
		const view = render(<Chip text={'Hello'} />);

		expect(view.getByText('Hello')).toBeDefined();
	});

	it('match snapshot', () => {
		const view = render(<Chip text={'Hello'} />);
		expect(view).toMatchSnapshot();
	});
});
