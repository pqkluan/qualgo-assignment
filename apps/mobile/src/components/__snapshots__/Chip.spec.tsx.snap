// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Chip match snapshot 1`] = `
<View
  style={
    {
      "backgroundColor": "#000000",
      "borderRadius": 4,
      "paddingHorizontal": 4,
      "paddingVertical": 2,
    }
  }
>
  <Text
    style={
      [
        {
          "color": "#000000",
        },
        {
          "fontSize": 14,
          "fontWeight": "400",
          "textAlign": "left",
        },
        {
          "color": "#FFFFFF",
        },
      ]
    }
  >
    Hello
  </Text>
</View>
`;
